install/migrations/update_10.0.x_to_11.0.0/form.php

```diff
@@ -401,3 +408,43 @@
     DAY_TIMESTAMP,
     param: 7
 );
+
+// Data format for ValidationField has changed
+if ($DB->tableExists('glpi_forms_destinations_formdestinations')) {
+    $iterator = $DB->request(['FROM' => 'glpi_forms_destinations_formdestinations']);
+    foreach ($iterator as $data) {
+        $config = json_decode($data['config'] ?? '[]', true) ?? [];
+
+        // If 'strategies' key is not set, skip this entry
+        // We probably already migrated it
+        if (!isset($config[ValidationField::getKey()]['strategies'])) {
+            continue;
+        }
+
+        $strategies = $config[ValidationField::getKey()]['strategies'] ?? [];
+        $specific_actors = $config[ValidationField::getKey()]['specific_actors'] ?? [];
+        $specific_question_ids = $config[ValidationField::getKey()]['specific_question_ids'] ?? [];
+        $specific_validationtemplates_ids = $config[ValidationField::getKey()]['specific_validationtemplates_ids'] ?? [];
+
+        $validation_configs = [];
+        foreach ($strategies as $strategy) {
+            $validation_configs[] = new ValidationFieldStrategyConfig(
+                strategy: ValidationFieldStrategy::tryFrom($strategy) ?? ValidationFieldStrategy::NO_VALIDATION,
+                specific_validationtemplate_ids: $specific_validationtemplates_ids,
+                specific_question_ids: $specific_question_ids,
+                specific_actors: $specific_actors,
+            );
+        }
+
+        $config[ValidationField::getKey()] = (new ValidationFieldConfig(
+            strategy_configs: $validation_configs
+        ))->jsonSerialize();
```

Comment:
Please try to use hardcoded strings/structures as much as possible in migrations. Indeed, if the `ValidationField::getKey()` changes in the future, or if the `ValidationFieldConfig::jsonSerialize()` produced structure changes, this migration will have its behaviour changed.